packages/nameguard-react/src/components/ReportIcon/index.tsx

```diff
@@ -56,19 +63,42 @@ export function ReportIcon({
   */
   ...props
 }: ReportShieldProps) {
-  const onClickHandler = () => {
-    if (onClickOverride) onClickOverride(ensName);
-    else {
-      window.location.href = getNameGuardURLForENSname(ensName);
+  const onClickHandler = (clickHandlerFor: ClickHandlerFor) => {
+    switch (clickHandlerFor) {
+      case ClickHandlerFor.icon:
+        if (onIconClickOverride) {
+          onIconClickOverride(ensName);
+        } else {
+          redirectUserToNameGuardInspectPage(ensName);
+        }
+        break;
+      case ClickHandlerFor.tooltip:
+        if (onTooltipClickOverride) {
+          onTooltipClickOverride(ensName);
+        } else {
+          redirectUserToNameGuardInspectPage(ensName);
+        }
+        break;
     }
   };
 
+  const redirectUserToNameGuardInspectPage = (ensName: ENSName) => {
```

Comment:
Seems we should move this to `/utils/url.ts` ? Please see my other comment with more details about that proposed file.